packages/chai-openapi-response-validator/lib/openapi-validator/lib/classes/OpenApi2Spec.js

```diff
@@ -2,10 +2,15 @@ const utils = require('../utils');
 const AbstractOpenApiSpec = require('./AbstractOpenApiSpec');
 const ValidationError = require('./errors/ValidationError');
 
-class OpenApi2Spec extends AbstractOpenApiSpec {
+class OpenApi2Spec extends AbstractOpenApiSpec { 
   findOpenApiPathMatchingPathname(pathname) {
+    const { basePath } = this.spec;
+    if(basePath && !pathname.startsWith(basePath)) {
+      throw new ValidationError('PATH_NOT_FOUND');
+    }
+    const endpoint = pathname.replace(basePath, '');
```

Comment:
I would probably reuse the code and naming conventions we use for OpenAPI 3, i.e.

```js
const pathnameWithoutBasePath = getPathnameWithoutBasePath(basePath, pathname),
    const openApiPath = utils.findOpenApiPathMatchingPossiblePathnames(
      [pathnameWithoutBasePath],
...
```
(`endpoint` can mean many things in a package about OpenAPI paths)

Reply:
Agreed. Changed `endpoint` to `pathnameWithoutBasePath` in the commit `Changed variable name for consistency`